demisto_sdk/commands/prepare_content/markdown_images_handler.py

```diff
@@ -0,0 +1,140 @@
+import os
+import re
+from pathlib import Path
+from urllib.parse import urlparse
+
+from demisto_sdk.commands.common.constants import (
+    GOOGLE_CLOUD_STORAGE_PUBLIC_BASE_PATH,
+    MARKDOWN_IMAGE_LINK_REGEX,
+    MARKDOWN_IMAGES_ARTIFACT_FILE_NAME,
+    SERVER_API_TO_STORAGE,
+    ImagesFolderNames,
+    MarketplaceVersions,
+    MarketplaceVersionToMarketplaceName,
+)
+from demisto_sdk.commands.common.handlers import JSON_Handler
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.commands.common.tools import get_file
+
+json = JSON_Handler()
+
+
+def replace_markdown_urls_and_upload_to_artifacts(
+    markdown_path: Path,
+    marketplace: MarketplaceVersions,
+    pack_name: str,
+    file_type: ImagesFolderNames,
+) -> dict:
+    """
+    This function goes over the pack readme.md file and by the marketplace value.
+    replaces the images url to the appropriate images urls.
+    Args:
+        markdown_path (Path): The path of the pack readme file
+        marketplace (MarketplaceVersions): The marketplace version
+        pack_name (str): The pack name
+    Returns:
+        - A dict in the form of {pack_name: [images_data]} or empty dict if no images urls were found in the README
+    """
+    readme_images_storage_data = (
+        collect_images_from_markdown_and_replace_with_storage_path(
+            markdown_path, pack_name, marketplace, file_type
+        )
+    )
+    # no external image urls were found in the readme file
+    if not readme_images_storage_data or not readme_images_storage_data[pack_name]:
+        logger.debug(f"no image links were found in {pack_name} readme file")
+        return {}
+
+    upload_markdown_images_to_artifacts(readme_images_storage_data, pack_name)
+    logger.info(f"{readme_images_storage_data=}")
+    return readme_images_storage_data
+
+
+def collect_images_from_markdown_and_replace_with_storage_path(
+    markdown_path: Path,
+    pack_name: str,
+    marketplace: MarketplaceVersions,
+    file_type: ImagesFolderNames,
+) -> dict:
+    """
+    Replaces inplace all images links in the pack README.md with their new gcs location
+
+    Args:
+        markdown_path (str): A path to the pack README file.
+        pack_name (str): A string of the pack name.
+        marketplace (str): The marketplace this pack is going to be uploaded to.
+
+    Returns:
+        A dict of the pack name and all the image urls found in the README.md file with all related data
+        (original_url - The original url as found in the README.md file.
+         final_dst_image_path - The destination where the image will be stored on gcp.
+         relative_image_path - The relative path (from the pack name root) in the gcp.
+         image_name - the image name)
+    """
+    google_api_readme_images_url = (
+        f"{GOOGLE_CLOUD_STORAGE_PUBLIC_BASE_PATH}/{MarketplaceVersionToMarketplaceName.get(marketplace)}"
```

Comment:
Let's replace this with a `GOOGLE_CLOUD_STORAGE_PUBLIC_MARKETPLACE_PATH` dict/Enum that should be added to constants.py - I'd rather we avoid having a nested non-explicit dependency of bucket-name<->URL - they should be in the same file.